### PR TITLE
Create rule S4423: add language AzureResourceManager

### PR DESCRIPTION
**Pull request by egon-okerman-sonarsource**
[Specification ticket](https://sonarsource.atlassian.net/browse/SONARIAC-755)
[Implementation ticket](https://sonarsource.atlassian.net/browse/SONARIAC-781)
[RSPEC Preview](https://sonarsource.github.io/rspec/#/rspec/S4423/azureresourcemanager)

Bicep PR for S4423: #1879

### CODE REVIEW

File: rules/S4423/azureresourcemanager/rule.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+----
+{
+  "$schema": "<https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",>
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "2022-09-01",
+      "name": "example",
+      // ...
```
Comment (mstachniuk) [inline rules/S4423/azureresourcemanager/rule.adoc:19]: IMO we can skip those 2 lines and the same in compliant code examples

File: rules/S4423/azureresourcemanager/rule.adoc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      "properties": {
+        "minimumTlsVersion": "TLS1_0" // Noncompliant
+      }
+    }
+  ]
+}
+----
+
```
Comment (mstachniuk) [inline rules/S4423/azureresourcemanager/rule.adoc:27]: Please also add examples in Bicep format and comment them for now.

File: rules/S4423/azureresourcemanager/rule.adoc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+----
+{
+  "$schema": "<https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",>
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.DBforMySQL/servers",
+      "apiVersion": "2017-12-01",
+      "name": "example",
+      // ...
```
Comment (mstachniuk) [inline rules/S4423/azureresourcemanager/rule.adoc:40]: Also, we can skip those 2 lines and line 42, and the same in compliant code examples

**Review by mstachniuk — CHANGES_REQUESTED** [review]
Please see my comments.

File: rules/S4423/azureresourcemanager/rule.adoc [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+include::../description.adoc[]
+
+=== Noncompliant code example
+
+For https://learn.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts[Azure Storage accounts], TLS 1.0 and 1.1 are accepted by default.
+
+[source,json,diff-id=2,diff-type=noncompliant]
+----
+{
+  "$schema": "<https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",>
```
[the remark this answers is not in the document]
Comment (egon-okerman-sonarsource) [reply rules/S4423/azureresourcemanager/rule.adoc:12]: Fixed but GH is a bit slow today

File: rules/S4423/azureresourcemanager/rule.adoc (hunk 2 above)
Comment (egon-okerman-sonarsource) [reply rules/S4423/azureresourcemanager/rule.adoc:27]: Bicep examples are kept in a separate branch: https://github.com/SonarSource/rspec/tree/rule/S4423/add-bicep. Do you think it is better to add it in comments here?

File: rules/S4423/azureresourcemanager/rule.adoc (hunk 1 above)
Comment (egon-okerman-sonarsource) [reply rules/S4423/azureresourcemanager/rule.adoc:19]: I added them to signify that there are some lines missing. There are some properties not set that are required.

Do you think it is worth adding those back, or would it just make the file too long?

File: rules/S4423/azureresourcemanager/rule.adoc [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "2022-09-01",
+      "name": "example",
+      "properties": {
+        "minimumTlsVersion": "TLS1_0" // Noncompliant
```
Comment (mstachniuk) [inline rules/S4423/azureresourcemanager/rule.adoc:20]: Comments in JSON are not allowed by syntax, please remove it

**Pull request merged**